frontend/src/js/components/subscription/SubscriptionPage.tsx

```diff
@@ -0,0 +1,393 @@
+type SelectedAddons = { [key in AvailableAddon]: boolean };
+
+type ContactReason = 'reduceLimit' | 'overLimit';
+interface ContactReasonProps {
+  reason: ContactReason;
+}
+const ContactReasonAlert = ({ reason }: ContactReasonProps) => {
+  const reasonMap: { [key in ContactReason]: ReactNode | string } = {
+    reduceLimit: `For over ${enterpriseDeviceCount} devices, please contact for pricing.`,
+    overLimit: (
+      <div>
+        If you want to reduce your device limit, please contact <SupportLink variant="email " />
+      </div>
+    )
+  };
+  return (
+    <Alert severity="info" className="margin-bottom-x-small margin-top-x-small">
+      {reasonMap[reason]}
+    </Alert>
+  );
+};
+
+export const SubscriptionPage = () => {
+  const [selectedPlan, setSelectedPlan] = useState<Plan>(PLANS.os);
+  const [selectedAddons, setSelectedAddons] = useState<SelectedAddons>({ configure: false, monitor: false, troubleshoot: false });
+  const [previewPrice, setPreviewPrice] = useState<PreviewPrice>();
+  const [order, setOrder] = useState();
+  const [contactReason, setContactReason] = useState<ContactReason | ''>('');
+  const [inputHelperText, setInputHelperText] = useState<string>(`The minimum limit for ${selectedPlan.name} is ${selectedPlan.minimalDeviceCount}`);
+  const [limit, setLimit] = useState<number>(selectedPlan.minimalDeviceCount);
+  const [enterpriseMessage, setEnterpriseMessage] = useState('');
+  const stripeAPIKey = useSelector(getStripeKey);
+  const [isPreviewLoading, setIsPreviewLoading] = useState(false);
+  const [showUpgradeDrawer, setShowUpgradeDrawer] = useState(false);
+  const [loadingFinished, setLoadingFinished] = useState(!stripeAPIKey);
+
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+  const currentDeviceLimit = useSelector(getDeviceLimit);
+  const hasCurrentPricing = useSelector(getHasCurrentPricing);
+  const org = useSelector(getOrganization);
+  const { addons: orgAddOns = [], plan: currentPlan = PLANS.os.id as AvailablePlans, trial: isTrial = true, id: orgId } = org;
+  const isOrgLoaded = !!orgId;
+  const plan = Object.values(PLANS).find(plan => plan.id === (isTrial ? PLANS.os.id : currentPlan)) || PLANS.os;
+  const enabledAddons = useMemo(() => orgAddOns.filter(addon => addon.enabled), [orgAddOns]);
+  const currentPlanId = plan.id;
+  const debouncedLimit = useDebounce(limit, TIMEOUTS.debounceDefault);
+
+  // redirect customers paying old price to billing page to see the alert
+  useEffect(() => {
+    if (org.id && !isTrial && !hasCurrentPricing) {
+      navigate('/settings/billing');
+    }
+  }, [isTrial, navigate, hasCurrentPricing, org]);
+
+  //Fetch Billing profile & subscription
+  useEffect(() => {
+    dispatch(getCurrentCard());
+    dispatch(getUserBilling());
+    dispatch(getUserSubscription());
+  }, [dispatch]);
```

Comment:
```suggestion
  useEffect(() => {
    if (isTrial) {
      return;
    }
    dispatch(getCurrentCard());
    dispatch(getUserBilling());
    dispatch(getUserSubscription());
  }, [dispatch, isTrial]);
```
as these calls will fail for trial users